### ISSUE TITLE
loss function for the person-in-wifi-3d dataset

### ISSUE THREAD

**Issue by aprilbian** [cseeyangchen/DT-Pose#4]
Dear Authors:

  Thanks for the great work! I am looking at the pose prediction task for the person-in-wifi-3d dataset and I am puzzled in line  214 and 215 of `train_pose.py` file where you wrote:
              `predicted_pose = torch.cat([predicted_pose[:,:num,:,:].reshape((-1, 14, 3)) for num in person_num], dim=0)`

  Note that this results in a predicted_pose of a huge dimension, e.g., when `person_num = [1, ..., 1]` and `batch_size = 32`, the first dimension of `predicted_pose` is 1024, yet it should be 32.

  In my understanding, the line should be written like this: `predicted_pose = torch.cat([predicted_pose[i,:person_num[i],:,:].reshape((-1, 14, 3)) for i in person_num.shape[0]], dim=0)`

  Could you kindly check if my understanding is correct? Thanks in advance for your reply!

**Comment by cseeyangchen** [comment]
Thanks for catching this! Your understanding is correct. The issue has been fixed. Specifically, `predicted_pose` and `pose_gt` in lines 214 and 215 have been updated to `predicted_pose = torch.cat([predicted_pose[idx,:num,:,:].reshape((-1, 14, 3)) for idx, num in enumerate(person_num)], dim=0)` and `pose_gt = torch.cat([pose_gt[idx,:num,:,:].reshape((-1, 14, 3)) for idx, num in enumerate(person_num)], dim=0)`. We have also applied the same fix to the corresponding testing code in lines 282 and 283.

**Comment by aprilbian** [comment]
> Thanks for catching this! Your understanding is correct. The issue has been fixed. Specifically, `predicted_pose` and `pose_gt` in lines 214 and 215 have been updated to `predicted_pose = torch.cat([predicted_pose[idx,:num,:,:].reshape((-1, 14, 3)) for idx, num in enumerate(person_num)], dim=0)` and `pose_gt = torch.cat([pose_gt[idx,:num,:,:].reshape((-1, 14, 3)) for idx, num in enumerate(person_num)], dim=0)`. We have also applied the same fix to the corresponding testing code in lines 282 and 283.


Thanks for your quick reply! So you think that this will not influence the prediction performance of the scheme reported in the paper? BTW, have you considered predicting the SMPL parameters instead of the keypoints? It would be great if I can hear from your opinion on the SMPL prediction.

**Comment by cseeyangchen** [comment]
> > Thanks for catching this! Your understanding is correct. The issue has been fixed. Specifically, `predicted_pose` and `pose_gt` in lines 214 and 215 have been updated to `predicted_pose = torch.cat([predicted_pose[idx,:num,:,:].reshape((-1, 14, 3)) for idx, num in enumerate(person_num)], dim=0)` and `pose_gt = torch.cat([pose_gt[idx,:num,:,:].reshape((-1, 14, 3)) for idx, num in enumerate(person_num)], dim=0)`. We have also applied the same fix to the corresponding testing code in lines 282 and 283.
> 
> Thanks for your quick reply! So you think that this will not influence the prediction performance of the scheme reported in the paper? BTW, have you considered predicting the SMPL parameters instead of the keypoints? It would be great if I can hear from your opinion on the SMPL prediction.

The one-person prediction results reported in the paper on the person-in-wifi-3d dataset are correct and were obtained using the original experimental code. The current repo contains a cleaned-up version, and the issue you pointed out was introduced during refactoring. It does not affect the results in the paper.

As for SMPL prediction, I believe it is more suitable than sparse keypoints, since WiFi signals are influenced by the full-body geometry rather than just sparse joints. However, the current WiFi datasets do not provide SMPL annotations, and converting these keypoints to SMPL adds extra IK ambiguity. I have tried to convert keypoints into the SMPL parameters using [smplify-x](https://github.com/cseeyangchen/smplifyx-skeleton), but I did not use these SMPL annotations for this WiFi-based pose estimation task. If SMPL parameters could be obtained using marker-based motion capture systems together with synchronized WiFi data, this would likely provide better supervision than those keypoints obtained from Kinect-based methods.
